ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FcsErrorsStageProcessBase.fs

```diff
@@ -106,8 +106,11 @@ type FcsErrorsStageProcessBase(fsFile, daemonProcess) =
 
         | NotAFunction ->
             let notAFunctionNode = nodeSelectionProvider.GetExpressionInRange(fsFile, range, false, null)
-            match (tryFindRootPrefixAppWhereExpressionIsFunc notAFunctionNode) with
-            | :? IPrefixAppExpr as prefixAppExpr -> NotAFunctionError(notAFunctionNode.IgnoreParentParens(), prefixAppExpr) :> _
+            let prefixApp =
+                if isNotNull notAFunctionNode then tryFindRootPrefixAppWhereExpressionIsFunc notAFunctionNode else null
```

Comment:
Wouldn't it be better to make tryFindRootPrefixAppWhereExpressionIsFunc allow getting/returning null? Maybe IgnoreParentParens is also would be better to have CanBeNull on both parameter and return.

Reply:
It's also interesting to note that related helpers (e.g. `IgnoreParentParens` for patterns) deals with null exactly like this).